homeassistant/components/switch/broadlink.py

```diff
@@ -141,10 +190,15 @@ def _get_mp1_slot_name(switch_friendly_name, slot):
 
     if switch_type in RM_TYPES:
         broadlink_device = broadlink.rm((ip_addr, 80), mac_addr)
-        hass.services.register(DOMAIN, SERVICE_LEARN + '_' +
-                               ip_addr.replace('.', '_'), _learn_command)
-        hass.services.register(DOMAIN, SERVICE_SEND + '_' +
-                               ip_addr.replace('.', '_'), _send_packet)
+        device_id = ip_addr.replace('.', '_')
+        if friendly_name:
```

Comment:
This will be a breaking change

Reply:
Yes it is.
but I think use `friendly_name` is better than a IP address
and it's better to manage multiple devices

Reply:
OK.  the docs updated.

Reply:
Wouldn't it be better to register a service handler, and access devices by entity_id, rather than creating multiple services, akin to what I'm doing here: https://github.com/cnrd/home-assistant/blob/db1940e912a7bc19cf43d76f6da46e87c4297f93/homeassistant/components/remote/xiaomi_miio.py#L101-L151 ?

Reply:
Yes,That's a very farsighted solution. 
But in my opinion, the sub Class just need is call this function,
there must be a centralized service register in super Class,
and we can get devices by entity_id.

Reply:
The code I'm linking is actually handling the enity_id -> device in the class and not in the remote superclass. If we decide to implement a general learn service in the superclass later it is easy to reuse the code.

Reply:
I like the suggestion from @cnrd.